maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireDependencyResolver.java

```diff
@@ -184,7 +189,17 @@ private ArtifactResolutionResult resolveArtifact( Artifact artifact, List<Artifa
             .setCollectionFilter( filter )
             .setRemoteRepositories( repositories );
 
-        return repositorySystem.resolve( request );
+        ArtifactResolutionResult result = repositorySystem.resolve( request );
+        try
+        {
+            resolutionErrorHandler.throwErrors( request, result );
+        }
+        catch ( ArtifactResolutionException e )
+        {
+            throw new RuntimeException( e );
```

Comment:
Pls throw `MojoExecutionException` instead.
Update method signatures.
Run `mvn clean` and reinstall the project again to make sure all modules are compilable.

Reply:
In https://github.com/apache/maven-surefire/pull/504/commits/8b4d3a51987a53ea83a0bdbfcc282418ae928996, PTAL.